File: backend/src/main/java/de/aivot/prosuna/backend/payment/models/PaymentStatus.java
package de.aivot.prosuna.backend.payment.models;

public enum PaymentStatus {
    PENDING,
    PAID,
    FAILED,
    CANCELED;

    public boolean isPending() {
        return this == PENDING;
    }

    public boolean isPaid() {
        return this == PAID;
    }

    public boolean isTerminal() {
        return this != PENDING;
    }
}

File: app/src/data/payment-status.ts
export enum PaymentStatus {
    Pending = 'PENDING',
    Paid = 'PAID',
    Failed = 'FAILED',
    Canceled = 'CANCELED',
}

export const PaymentStatusLabels: Record<PaymentStatus, string> = {
    [PaymentStatus.Pending]: 'Zahlung ausstehend',
    [PaymentStatus.Paid]: 'Bezahlt',
    [PaymentStatus.Failed]: 'Bezahlung fehlgeschlagen',
    [PaymentStatus.Canceled]: 'Zahlung abgebrochen',
};
